presto-main/src/main/java/com/facebook/presto/sql/planner/PlanBuilder.java

```diff
@@ -92,6 +97,11 @@ public TranslationMap getTranslations()
 
     public PlanBuilder appendProjections(Iterable<Expression> expressions, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
     {
+        return appendProjections(expressions, symbolAllocator, idAllocator, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
```

Comment:
I was rather thinking that you could just call:
```
appendProjections(groupingOperationRewriter.rewrite(groupingExpression), symbolAllocator, planNodeIdAllocator)
```
instead of modifying those methods.

You are right that `appendProjections` the code tries to get the type of the expression. Since that is the case, then I think the copy paste of logic is OK in `handleGroupings` method